augur/frequency_estimators.py

```diff
@@ -869,82 +871,52 @@ def to_json(self):
         return frequencies_json
 
     @classmethod
-    def calculate_pivots(cls, pivot_frequency, tree=None, start_date=None, end_date=None):
-        """
-        Calculate pivots for a given pivot frequency and either a tree or a start and end date.
+    def calculate_pivots(cls, pivot_frequency, observations=None, start_date=None, end_date=None):
+        """Calculate pivots for a given pivot frequency and either a list of observation
+        dates or a start and end date.
 
-        If a tree is given, the start and end interval for these pivots is
-        determined by the earliest and latest strain date in the tree.
+        If a list of observations is given, the start and end interval for these
+        pivots is determined by the earliest and latest strain date in the list.
 
-        If a start and end date are given, those values determine the range
-        of the pivots. These values and the tree are mutually exclusive.
-        If all arguments are provided, the start and end dates will be
-        preferred over the tree.
+        If a start and end date are given, those values determine the range of
+        the pivots. These values and the list of observations are mutually
+        exclusive. If all arguments are provided, the start and end dates will
+        be preferred over the list of observations.
 
-        Parameters
-        ----------
-        pivot_frequency
-            float
-            frequency pivots should occur by fraction of a year
-        tree
-            Bio.Phylo
-            an annotated tree
-        start_date
-            float
-            start of the pivots interval
-        end_date
-            float
-            end of the pivots interval
-        -----------
+        Args:
+            pivot_frequency (int): number of months between pivots
+            observations (list): a list of observed floating point dates per sample
+            start_date (float): start of the pivots interval
+            end_date (float): end of the pivots interval
+
+        Returns:
+            pivots (numpy array): pivots spanning the given the dates
 
-        Returns
-        -------
-        --------
-            pivots (numpy array): pivots spanning the given the dates represented
-            by the tree's tips
         """
-        if start_date is None or end_date is None:
-            # Determine pivot start and end dates from the range of tip dates in the given tree.
-            tip_dates = [tip.attr["num_date"] for tip in tree.get_terminals()]
-            pivot_start = min(tip_dates)  # type: float
-            pivot_end = max(tip_dates)    # type: float
-        else:
+        if start_date and end_date:
```

Comment:
why require both start and end date and overwrite if only one is specified?

Reply:
Fixed!